lib/engine/tile.rb

```diff
@@ -395,6 +395,14 @@ def city_town_edges
       ct_edges.values
     end
 
+    def city_town_edges_are_subset_of?(other_cte)
+      cte = city_town_edges
+      ALL_EDGES.any? do |rotation|
```

Comment:
this seems expensive what are you trying to do?

Reply:
I am checking that each city has a matching city (in terms of edge connections) on the new tile instead of simply the whole tile having matching edges. To try and mitigate the expense I had made sure this was only called for tiles with more than one city, rather than when checking every upgrade. Do you think it's still an issue given this is the case?

(The only alternative route I can see to fixing the issue is to move all the code for checking a valid rotation out of the base Tracker step into the base Game definition, and thereby give the Tiles view access to those checks, so it can check for a valid rotation. This however will require that I touch every game that has overriden any of those methods in the track-laying step or which has some tiles with only certain rotations allowed. I can do this but it's obviously a much bigger change to the codebase.)